src/models/DRBResult.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+  DRBWork,
+  Edition,
+  Author,
+  Agent,
+  EditionLink,
+} from "../types/drbTypes"
+import { SOURCE_PARAM, DRB_BASE_URL } from "../config/constants"
+import { readOnlineMediaTypes, downloadMediaTypes } from "../utils/drbUtils"
+/**
+ * The DRBResult class contains the data and getter functions
+ * for a single DRB search result entity.
+ *
+ * DRB Results returned from the API are mapped and initialized into
+ * DRBResult objects in the DRBContainer component.
+ */
+export default class DRBResult {
+  id: string
+  title: string
+  editions?: Edition[]
+  authors?: Author[] | Agent[]
+
+  constructor(work: DRBWork) {
+    this.id = work.uuid
+    this.title = work.title
+    this.editions = work.editions
+    this.authors = this.getAuthorsFromWork(work)
+  }
+
+  get url(): string {
+    return `${DRB_BASE_URL}/work/${this.id}${SOURCE_PARAM}`
+  }
+
+  get selectedEdition(): Edition {
+    return (
+      this.editions.find(
+        (edition) => edition?.items?.length && edition.items[0].links.length
+      ) || this.editions[0]
+    )
+  }
+
+  get readOnlineUrl(): string | null {
+    if (!this.selectedEdition.items) return null
+
+    // Populate selected link with first item on the selected edition that includes
+    // a link with a media type included in the readOnlineMediaTypes array.
+    let selectedLink: EditionLink
+    this.selectedEdition.items.find((item) =>
+      item.links.find((link) => {
+        selectedLink = link
```

Comment:
Interesting. `.find` returns the first found object so would this also work if you assign `this.selectedEdition.items....` to `selectedLink`? This is more out of curiosity because the way this breaks and sets the selected link object is interesting.

Reply:
The value can be assigned from the return value of the inner find(), but not the outer one, since it returns the item instead of the link.

Just moved the assignment to the inner find to avoid having to assign the value from within the matcher function.